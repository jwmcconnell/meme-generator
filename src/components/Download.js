import React from 'react';
import PropTypes from 'prop-types';

const Download = ({ downloadMeme }) => {
  return <button onClick={downloadMeme}>Download</button>;
};

Download.propTypes = {
  downloadMeme: PropTypes.func.isRequired
};

export default Download;
