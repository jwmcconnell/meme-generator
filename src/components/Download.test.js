/* eslint-disable no-console */
import React from 'react';
import { shallow } from 'enzyme';
import Download from './Download';

describe('Download component', () => {
  it('renders Download', () => {
    
    const downloadMeme = e => console.log(e.target.value);
    
    const wrapper = shallow(
      <Download 
        downloadMeme={downloadMeme}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
  
