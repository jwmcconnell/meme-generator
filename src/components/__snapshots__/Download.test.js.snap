// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download component renders Download 1`] = `
<button
  onClick={[Function]}
>
  Download
</button>
`;
